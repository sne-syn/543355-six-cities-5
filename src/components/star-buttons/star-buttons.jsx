import PropTypes from 'prop-types';
import React from 'react';
import {MAX_RATING} from '../../utils/const';
import {connect} from 'react-redux';
import {getReviewRating} from '../../store/reviews/reviews-selectors';
import {setReviewRating} from '../../store/action';

const starButtonsTitle = {
  5: `perfect`,
  4: `good`,
  3: `not bad`,
  2: `badly`,
  1: `terribly`,
};

const StarButtons = ({onClickAction}) => {
  return (
    <div className="reviews__rating-form form__rating">
      {[...Array(MAX_RATING)].map((_star, i) => {
        return (
          <React.Fragment key={i}>
            <input className="form__rating-input visually-hidden" name="rating" value={MAX_RATING - i} id={`${MAX_RATING - i}-star`} type="radio" onClick={onClickAction}/>
            <label htmlFor={`${MAX_RATING - i}-star`} className="reviews__rating-label form__rating-label" title={starButtonsTitle[+MAX_RATING - i]}>
              <svg className="form__star-image" width="37" height="33">
                <use xlinkHref="#icon-star"></use>
              </svg>
            </label>
          </React.Fragment>
        );
      }
      )}
    </div>
  );
};

StarButtons.propTypes = {
  onClickAction: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  rating: getReviewRating(state),
});

const mapDispatchToProps = (dispatch) => ({
  onClickAction(evt) {
    dispatch(setReviewRating(evt));
  },
});

export {StarButtons};
export default connect(mapStateToProps, mapDispatchToProps)(StarButtons);
