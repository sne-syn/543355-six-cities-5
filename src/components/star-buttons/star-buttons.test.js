import React from 'react';
import renderer from 'react-test-renderer';
import {StarButtons} from './star-buttons';

const noop = () => {};
test(`Render StarButtons`, () => {
  const tree = renderer
    .create(<StarButtons 
      onChange={noop}
      onStarClick={noop}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
