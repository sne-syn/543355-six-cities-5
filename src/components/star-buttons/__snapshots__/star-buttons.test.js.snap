// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render StarButtons 1`] = `
<div
  className="reviews__rating-form form__rating"
>
  <input
    className="form__rating-input visually-hidden"
    id="5-star"
    name="rating"
    onClick={[Function]}
    type="radio"
    value={5}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-star"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    className="form__rating-input visually-hidden"
    id="4-star"
    name="rating"
    onClick={[Function]}
    type="radio"
    value={4}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="4-star"
    title="good"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    className="form__rating-input visually-hidden"
    id="3-star"
    name="rating"
    onClick={[Function]}
    type="radio"
    value={3}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="3-star"
    title="not bad"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    className="form__rating-input visually-hidden"
    id="2-star"
    name="rating"
    onClick={[Function]}
    type="radio"
    value={2}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="2-star"
    title="badly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    className="form__rating-input visually-hidden"
    id="1-star"
    name="rating"
    onClick={[Function]}
    type="radio"
    value={1}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-star"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
</div>
`;
